[BUG] Exhausted input when reading schema

I've been observing this error for a while, and even though its transient, I think the way it should be handled is via ZIO retry rather than container restart. Container restart creates a lookback and if we are unlucky enough to read empty file often, it can take a while before we can get fresh data merged.

```
	at ujson.StringParser$.transform(StringParser.scala:28)
	at ujson.Readable$fromTransformer.transform(Readable.scala:19)
	at upickle.Api.read$$anonfun$1(Api.scala:42)
	at upickle.core.TraceVisitor$.withTrace(TraceVisitor.scala:18)
	at upickle.Api.read(Api.scala:42)
	at upickle.Api.read$(Api.scala:16)
	at upickle.default$.read(Api.scala:179)
	at com.sneaksanddata.arcane.framework.models.cdm.SimpleCdmModel$.apply(SimpleCdmModel.scala:64)
	at com.sneaksanddata.arcane.framework.services.cdm.CdmSchemaProvider.getEntity$$anonfun$3$$anonfun$2(CdmSchemaProvider.scala:36)
	at zio.ZIO.map$$anonfun$1$$anonfun$1(ZIO.scala:972)
	at com.sneaksanddata.arcane.framework.services.cdm.CdmSchemaProvider.getEntity(CdmSchemaProvider.scala:36)
	at com.sneaksanddata.arcane.framework.services.cdm.CdmSchemaProvider.getEntity(CdmSchemaProvider.scala:37)
	at com.sneaksanddata.arcane.framework.services.cdm.CdmSchemaProvider.getSchema(CdmSchemaProvider.scala:29)
	at com.sneaksanddata.arcane.microsoft_synapse_link.services.data_providers.microsoft_synapse_link.CdmTableStream.enrichWithSchema(CdmTableStream.scala:123)
Caused by: ujson.IncompleteParseException: exhausted input
```

--- a/src/main/scala/services/data_providers/microsoft_synapse_link/CdmTableStream.scala
+++ b/src/main/scala/services/data_providers/microsoft_synapse_link/CdmTableStream.scala
@@ -120,9 +120,14 @@ class CdmTableStream(name: String,
   private def enrichWithSchema(stream: ZStream[Any, Throwable, StoredBlob]): ZStream[Any, Throwable, SchemaEnrichedBlob] =
     stream.filterZIO(prefix => azureBlogStorageReader.blobExists(storagePath + prefix.name + "model.json"))
       .mapZIO(prefix =>
+
+        val task = CdmSchemaProvider(reader, (storagePath + prefix.name).toHdfsPath, name)
+          .getSchema
+          .tapErrorCause(e => zlog(s"Failed to read schema from: ${prefix.name}")).retry(retryPolicy)
+
         for
           promise <- Promise.make[Throwable, ArcaneSchema]
-          fiber <- promise.complete(CdmSchemaProvider(reader, (storagePath + prefix.name).toHdfsPath, name).getSchema.retry(retryPolicy)).fork
+          fiber <- promise.complete(task).fork
           _ <- fiber.join
         yield SchemaEnrichedBlob(prefix, promise)
     )

